behat/tests/bootstrap/UserTrait.php

```diff
@@ -16,6 +18,35 @@ trait UserTrait
         'prof named' => 'ROLE_PROF_NAMED'
     ];
 
+    /**
+     * Requires an authenticated admin user to use in a scenario
+     *
+     * @Given the following admins exist:
+     */
+    public function adminsExist(TableNode $table)
+    {
+        foreach ($table as $inputs) {
+            $this->assertValidAdminRole($inputs['adminType']);
+
+            $adminType = $inputs['adminType'];
+            $firstName = $inputs['firstName'];
+            $lastName = $inputs['lastName'];
+            $email = $inputs['email'];
+            $activated = $inputs['activated'];
+
+            $query = "adminType=$adminType&firstName=$firstName&lastName=$lastName&email=$email&activated=$activated";
```

Comment:
I believe this could just be:
```php
$query = http_build_query($inputs);
```
Though that doesn't sanitise the query keys, so test-writers could add anything they like to the query string.

Reply:
Much nicer - I've expanded the admin type assertion to check for the required keys too 👌